packages/gatsby-transformer-sharp/src/customize-schema.js

```diff
@@ -374,6 +379,182 @@ const fluidNodeType = ({
   }
 }
 
+const imageNodeType = ({
+  pathPrefix,
+  getNodeAndSavePathDependency,
+  reporter,
+  cache,
+}) => {
+  return {
+    type: new GraphQLObjectType({
+      name: `ImageSharpGatsbyImage`,
+      fields: {
+        imageData: {
+          type: new GraphQLNonNull(GraphQLJSON),
+        },
+      },
+    }),
+    args: {
+      layout: {
+        type: ImageLayoutType,
+        defaultValue: `fixed`,
+        description: stripIndent`
+        The layout for the image.
+        FIXED: A static image sized, that does not resize according to the screen width
+        FLUID: The image resizes to fit its container. Pass a "sizes" option if it isn't going to be the full width of the screen. 
+        CONSTRAINED: Resizes to fit its container, up to a maximum width, at which point it will remain fixed in size.
+        `,
+      },
+      maxWidth: {
+        type: GraphQLInt,
+        description: stripIndent`
+        Maximum display width of generated files. 
+        The actual largest image resolution will be this value multipled by the largest value in outputPixelDensities
+        This only applies when layout = FLUID or CONSTRAINED. For other layout types, use "width"`,
+      },
+      maxHeight: {
+        type: GraphQLInt,
+      },
+      width: {
+        type: GraphQLInt,
+        description: stripIndent`
+        The display width of the generated image. 
+        The actual largest image resolution will be this value multipled by the largest value in outputPixelDensities
+        Ignored if layout = FLUID or CONSTRAINED, where you should use "maxWidth" instead.
+        `,
+      },
+      height: {
+        type: GraphQLInt,
+      },
+      placeholder: {
+        type: ImagePlaceholderType,
+        defaultValue: `blurred`,
+        description: stripIndent`
+        Format of generated placeholder image. 
+        DOMINANT_COLOR: a solid color, calculated from the dominant color of the image. 
+        BASE64: a blurred, low resolution image, encoded as a base64 data URI
+        TRACED_SVG: a low-resolution traced SVG of the image.
+        NONE: no placeholder. Set "background" to use a fixed background color.`,
+      },
+      tracedSVGOptions: {
+        type: PotraceType,
+        defaultValue: false,
+        description: `Options for traced placeholder SVGs. You also should set placeholder to SVG.`,
+      },
+      webP: {
+        type: GraphQLBoolean,
+        defaultValue: true,
+        description: `Generate images in WebP format as well as matching the input format. This is the default (and strongly recommended), but will add to processing time.`,
+      },
```

Comment:
Any reason why we want this option? :) I don't immediately see the need to disable it.

Reply:
It doubles the processing time, so some sites where that's a big problem might want to

Reply:
(Or maybe not doubles, but increases signficantly)

Reply:
The biggest reason why I wouldn't want people to disable it in here is that they value DX over UX. With lazy images and query-on-demand, this becomes less of an issue IMO but I do get your point.

What if we change the description to talk more about performance and not the built time.
```
description: `Generate images in WebP format as well as matching the input format. This is the default (and strongly recommended), it improves your website performance.`,
```

Reply:
What if the default was `false` during `gatsby develop` (i.e. NODE_ENV === development) for better DX perf, but `true` during `gatsby build` (i.e. NODE_ENV === production) for better UX?

Reply:
we'll fix this with lazy images https://github.com/gatsbyjs/gatsby/discussions/27603

Reply:
Build time is still important in production. My last job had a site that wouldn't build within the Netlify time limit if we had WebP enabled. We moved it to AWS, but it was still something we needed to be able to disable.

My plan for the new API will be to allow the user to pass an array of formats, with the default being `[AUTO, WEBP]`

Reply:
Build times are never more important than users. That's why I want this description to be different.